editor.planx.uk/src/@planx/components/List/model.ts

```diff
@@ -42,6 +42,7 @@ export type NumberField = {
 export type QuestionField = {
   type: "question";
   required?: boolean;
+  unique?: boolean;
```

Comment:
The value of a real example in spotting oversights!

This reminded me that one of the restrictions we discussed would be that certain fields could be unique - meaning only one item in the list could contain this value.

Reply:
This is a super good spot & a tricky one to consider! A few open questions: 
- I like the visual disabled feedback above, and it made me wonder if when there is an item in the schema set as "unique", should we also think of future validation that the "max" can't exceed the number of items in the unique field - otherwise you could add an item but the select could be entirely disabled? (Wouldn't necessarily make sense if "unique" ever has to be applied outside of `QuestionField` tho?)
- Can _many_ fields within the same schema have a unique constraint? Again, interesting if we ever want to try to automatically set "max" (calculate possible permutations?!)

Reply:
Yep! Hit both these exact questions also 💡 

I think right now...
 - We have no form validation for creation of `Schema` models, so can't enforce max === unique length natively in TS, we have to trust Editor's judgements to set max correctly. Eventually we'd enforce this via the Editor modal form that makes the schema. However, this actually isn't a real issue right now as if you have 5 items in a `unique` question and add a 6th list item, you can't save or proceed - so it's sort of self-limiting at least?
 - Yep - super interesting, and we start hitting permutation questions of a unique constraint on the _combination_ of question A and question B. I think this is actually possible and not super hard (again, we can calculate a max dynamically - maybe a better approach than the above "trust Editors" approach). I think this is relatively easily enforced via the UI, trickier to generate a dynamic yup schema which does this. 
 
Personal opinion is that either or both of these are sensible potential next steps, but maybe not day 1 requirements (likely a consideration for the content team to decide).